### DISCUSSION TITLE
Why is bokeh not in package dependencies

### DISCUSSION THREAD

**Discussion by gabicca** [dask/distributed#7359] [Q&A] [answered]
Hi, I was wondering why bokeh is not part of the package requirements if it's quite essential for distributed? Since it's not in requirements.txt, it's also not in setup.py. So when you install the package, bokeh is not installed, which causes errors when parts of distributed is used.

**Comment by jacobtomlinson** [comment, marked as answer]
Bokeh isn't really essential for distributed. You only need it if you want to look at the dashboard. Many folks don't need it (especially in production environments) and also care about keeping their environments as small as possible.

Recently it was accidentally made non-optional and that broke things for a bunch of folks so it was restored back to optional in #7230.

When bokeh is missing you **should** see nice warning messages saying if you want to use that feature you need to install bokeh. If you're seeing hard crashes and tracebacks those should be fixed.

**Reply by gabicca** [reply]
Thanks for the answer.

Yes, I'm seeing an error:
```
/usr/local/lib/python3.10/dist-packages/distributed/client.py:4615: in _get_task_stream
    from distributed.dashboard.components.scheduler import task_stream_figure
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    from __future__ import annotations
    
    import weakref
    
>   from bokeh.core.properties import without_property_validation
E   ModuleNotFoundError: No module named 'bokeh'
/usr/local/lib/python3.10/dist-packages/distributed/dashboard/components/__init__.py:5: ModuleNotFoundError
```

If bokeh is optional, there should be a way of installing distributed with the extra dependency (in setup.py `extras_require`). Since bokeh is not a dependency of my package I shouldn't be including it in my requirements.

I also raised this: https://github.com/dask/distributed/issues/7173#issuecomment-1323693251 Where even if we have bokeh but not the right version, it crashes instead of just warn as it was intended to.

**Reply by jacobtomlinson** [reply]
Could you open that as an issue? We should fix up that error. And maybe another one for the `extras_require`?

**Reply by gabicca** [reply]
All right, I'll open the two issues. Thanks again!

**Reply by gabicca** [reply]
https://github.com/dask/distributed/issues/7362

and 

https://github.com/dask/distributed/issues/7363
